### PR TITLE
[#26521][manager] gem whenever to invite guest

### PR DESCRIPTION
**Pull request by anhdtn-1027**

### CODE REVIEW

**Comment by ngocAnh1999** [comment]
image

**Comment by ngocAnh1999** [comment]
link edu-redmine: https://edu-redmine.sun-asterisk.vn/issues/26521

**Comment by ngocAnh1999** [comment]
ready

**Comment by anhdtn-1027** [comment]
ready

File: app/workers/invite_guest_whenever_worker.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class InviteGuestWheneverWorker
+  include Sidekiq::Worker
+
+  def perform
+    day = Time.zone.now
+    events = Event.includes(:room).activate.where("rooms.active = TRUE").references(:room)
```
Comment (ngocvt-0484) [inline app/workers/invite_guest_whenever_worker.rb:6]: gọi scope e nhé

File: app/workers/invite_guest_whenever_worker.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    events.each do |event|
+      next unless valid_day?(event.start_time, day) && valid_time?(event.start_time, day)
+
+      users = User.includes(guests: :event).where("events.id = ?", event.id).references :guests
```
Comment (ngocvt-0484) [inline app/workers/invite_guest_whenever_worker.rb:10]: use scope

File: config/schedule.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# end
+#
+# every 4.days do
+#   runner "AnotherModel.prune_old_records"
+# end
+
+# Learn more: http://github.com/javan/whenever
+
+every 30.minute do # Many shortcuts available: :hour, :day, :month, :year, :reboot
+  rake "batch:invite_messages", environment: "development"
```
Comment (ngocvt-0484) [inline config/schedule.rb:26]: đừng set enviroment e
sau staging vs productiont thì chạy ntn

File: app/workers/invite_guest_whenever_worker.rb (hunk 1 above)
Comment (anhdtn-1027) [reply app/workers/invite_guest_whenever_worker.rb:6]: vâng ạ

File: app/workers/invite_guest_whenever_worker.rb (hunk 2 above)
Comment (anhdtn-1027) [reply app/workers/invite_guest_whenever_worker.rb:10]: vâng :D

File: config/schedule.rb (hunk 3 above)
Comment (anhdtn-1027) [reply config/schedule.rb:26]: vâng ạ

**Comment by anhdtn-1027** [comment]
ready

**Pull request merged**